tests/functional/nar-access.sh

```diff
@@ -25,6 +25,12 @@ diff -u baz.cat-nar $storePath/foo/baz
 nix store cat $storePath/foo/baz > baz.cat-nar
 diff -u baz.cat-nar $storePath/foo/baz
 
+# Check that 'nix store cat' fails on invalid store paths.
+invalidPath="$(dirname $storePath)/99999999999999999999999999999999-foo"
+mv $storePath $invalidPath
+(! nix store cat $invalidPath/foo/baz)
```

Comment:
```suggestion
expect 1 nix store cat $invalidPath/foo/baz
```
(or whatever the exit code is)